tests/test_simresults.py

```diff
@@ -152,12 +152,12 @@ def test_get_final_state_noisy():
     assert isdiagonal(final_state)
     res3._meas_basis = "ground-rydberg"
     assert (
-        final_state[0, 0] == 0.06666666666666667 + 0j
-        and final_state[2, 2] == 0.92 + 0j
+        final_state[0, 0] == 0.12 + 0j
+        and final_state[2, 2] == 0.8666666666666667 + 0j
     )
     assert res3.states[-1] == final_state
     assert res3.results[-1] == Counter(
-        {"10": 0.92, "00": 0.06666666666666667, "11": 0.013333333333333334}
+        {"10": 0.8666666666666667, "00": 0.12, "11": 0.013333333333333334}
     )
```

Comment:
Woaw that's weird to have these floats hardcoded and checking against them like that.
I had no idea we do this.

What about using `math.isclose()`?

Reply:
I'm not a fan either, but it's important to have some fixed values to catch changes like this.

How would you use `math.isclose()` when comparing `Counter`s?

Reply:
I understand these are hard guards providing strong guarantees, but float equality is touchy, so I am unsure about the lifespan of such guarantees. It may even be architecture dependent regarding the underlying float type.

Probably I would loop over the keys of the `Counter`.

That being said, this is outside the scope of this PR. And since it has been working until now on many many machines, I see no real reasons to address it.